tests/lit-tests/signed-integer-overflow.ispc

```diff
@@ -0,0 +1,186 @@
+
+unmasked int16 add_16(int16 a, int16 b) { return a + b; }
+
+unmasked int8 add_8(int8 a, int8 b) { return a + b; }
+
+
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i64> @add_u64
+// CHECK_NSW-NOT:             %{{.*}}= add{{.*}}nsw{{.*}}i64
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i32> @add_u32
+// CHECK_NSW-NOT:             %{{.*}}= add{{.*}}nsw{{.*}}i32
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i16> @add_u16
+// CHECK_NSW-NOT:             %{{.*}}= add{{.*}}nsw{{.*}}i16
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i8> @add_u8
+// CHECK_NSW-NOT:             %{{.*}}= add{{.*}}nsw{{.*}}i8
+
+unmasked uint64 add_u64(uint64 a, uint64 b) { return a + b; }
+
+unmasked uint32 add_u32(uint32 a, uint32 b) { return a + b; }
+
+unmasked uint16 add_u16(uint16 a, uint16 b) { return a + b; }
+
+unmasked uint8 add_u8(uint8 a, uint8 b) { return a + b; }
+
+
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i64> @sub_64
+// CHECK_NSW-COUNT-1:         %{{.*}}= sub{{.*}}nsw{{.*}}i64
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i32> @sub_32
+// CHECK_NSW-COUNT-1:         %{{.*}}= sub{{.*}}nsw{{.*}}i32
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i16> @sub_16
+// CHECK_NSW-COUNT-1:         %{{.*}}= sub{{.*}}nsw{{.*}}i16
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i8> @sub_8
+// CHECK_NSW-COUNT-1:         %{{.*}}= sub{{.*}}nsw{{.*}}i8
+
+unmasked int64 sub_64(int64 a, int64 b) { return a - b; }
+
+unmasked int32 sub_32(int32 a, int32 b) { return a - b; }
+
+unmasked int16 sub_16(int16 a, int16 b) { return a - b; }
+
+unmasked int8 sub_8(int8 a, int8 b) { return a - b; }
+
+
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i64> @sub_u64
+// CHECK_NSW-NOT:             %{{.*}}= sub{{.*}}nsw{{.*}}i64
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i32> @sub_u32
+// CHECK_NSW-NOT:             %{{.*}}= sub{{.*}}nsw{{.*}}i32
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i16> @sub_u16
+// CHECK_NSW-NOT:             %{{.*}}= sub{{.*}}nsw{{.*}}i16
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i8> @sub_u8
+// CHECK_NSW-NOT:             %{{.*}}= sub{{.*}}nsw{{.*}}i8
+
+unmasked uint64 sub_u64(uint64 a, uint64 b) { return a - b; }
+
+unmasked uint32 sub_u32(uint32 a, uint32 b) { return a - b; }
+
+unmasked uint16 sub_u16(uint16 a, uint16 b) { return a - b; }
+
+unmasked uint8 sub_u8(uint8 a, uint8 b) { return a - b; }
+
+
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i64> @shl_64
+// CHECK_NSW-COUNT-1:         %{{.*}}= shl{{.*}}nsw{{.*}}i64
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i32> @shl_32
+// CHECK_NSW-COUNT-1:         %{{.*}}= shl{{.*}}nsw{{.*}}i32
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i16> @shl_16
+// CHECK_NSW-COUNT-1:         %{{.*}}= shl{{.*}}nsw{{.*}}i16
+// CHECK_NSW-LABEL:         define <{{[0-9]*}} x i8> @shl_8
+// CHECK_NSW-COUNT-1:         %{{.*}}= shl{{.*}}nsw{{.*}}i8
+
+unmasked int64 shl_64(int64 a, int64 b) { return a << b; }
```

Comment:
`nsw` should not be set on shift operations.

Reply:
Why not? According the the manual, it accepts [NSW](https://llvm.org/docs/LangRef.html#shl-instruction).

Reply:
The result can change sign if shifted by an amount that's too large, causing signed overflow. https://groups.google.com/g/llvm-dev/c/TbbEn6kJkWQ

Reply:
The patch has been generating and testing shifts for left shifts for some time now. Why is this coming up for the first time now?

Reply:
https://godbolt.org/z/ETWdT3Wdo - no `nsw` / `nuw` bit observed.

> Why is this coming up for the first time now?
It's good that we were able to catch it before it was committed.

Reply:
Will revert the change for Shl and stdlib. Still, it would've been better to have given the feedback when discussing design requirements and when [reviewing the test](https://github.com/ispc/ispc/pull/2494#discussion_r1190462995) for NSW on the basic math operators.

Reply:
Done. CI is green.